packages/ai-chat/src/browser/ai-chat-preferences.ts

```diff
@@ -38,6 +39,14 @@ If no Default Agent is configured, Theia´s defaults will be applied.'),
 You can manually unpin or switch agents anytime.'),
             default: true,
             title: AI_CORE_PREFERENCES_TITLE,
+        },
```

Comment:
We should move this pref to workspace-preferences.ts in the ide package. It only makes sense if tool have a workspace. Furthermore, I would move them under the "promptTemplates" category. They are related to prompt fragments.

Reply:
The `ai-chat` package already depends on the `workspace` package, so there's no need to move this, at least from the perspective of dependency management. I'm willing to move it, but I don't really like having different pieces of the same functionality spread out if they don't have to be.

Reply:
It's not about dependency management. It is more about keeping the core clean from IDEish or more specific things. And having a workspace relative directory for taskContexts is something that probably many tool would not want.

For this reason, I would turn around the flow. The Task-Context-Service has a "setter" for a location. The IDE package can set this to a workspace related directory if the pref is set. So the task context service interface is clean and the task context service does not know the user pref.

Adopters can then choose to set a static directory for example or do a custom pref.

Reply:
I have a little bit of an aversion to setters - harder to figure out who's setting what when - so I've added an additional interface for `TaskContextStorageService` with the default implementation being the in-memory version and then an additional preference+file-based version in the `ai-ide` package. How does that sound?

Reply:
that sounds very good to me!